pkg/controllers/osc/osc_controller.go

```diff
@@ -209,7 +212,9 @@ func (r *Reconciler) reconcileOperatingSystemConfigs(ctx context.Context, md *cl
 
 func (r *Reconciler) reconcileSecrets(ctx context.Context, md *clusterv1alpha1.MachineDeployment) error {
 	oscList := &osmv1alpha1.OperatingSystemConfigList{}
-	if err := r.List(ctx, oscList, &client.ListOptions{Namespace: r.namespace}); err != nil {
+	if err := r.List(ctx, oscList, &client.ListOptions{Namespace: r.namespace}, client.MatchingLabels{
+		resources.MachineDeploymentSelector: fmt.Sprintf("%s-%s", md.Namespace, md.Name),
+	}); err != nil {
```

Comment:
Assuming we'll have only one `OSC` for now we can simply generate the OSC name like https://github.com/kubermatic/operating-system-manager/blob/master/pkg/controllers/osc/osc_controller.go#L281 and then retrieve that object.

Reply:
I agree it would be a simpler solution, but it would require a further small change also in the following lines. It works as expected and allows having more than a single OSC for md. Do you think that forcing to use a single OSC would be better so far? Personally, I don't see any drawback in keeping this implementation.

Reply:
>It works as expected and allows having more than a single OSC for md

Right now there is a 1-to-1 mapping between a md and OSC/secret. I don't think there will be a use case where we have multiple OSC atm. It's completely up to you since it's not important but I'll recommend just sticking with a single OSC for now. Instead of using labels to list down the OSCss.

Obviously in the future md and OSC will have a 1-to-n relationship.